ui: database details page now supports a large number of tables

j82w opened the pull request:
Previous design:
1. The ui would attempt to a network call for all tables in a db.
2. The ui would completely fail if more than roughly 900 tables were present in a db. The number could be less if the table names were long.

New design:
1. The ui will only load the table details of the first 2 pages. This is a currently set to 20 tables per a page. If more than 40 tables are present in a db, the ui will disable sort. This is necessary because it would require loading the details of all tables which results in to many network calls overloading the db. The user can still filter using the table name.
2. The ui will now load all the table names. If a max size error is hit the ui will loop until it pulls all the table names.

Fixes: #98819
Epic: none

Before: https://www.loom.com/share/8b63f3ad96e74269be6a8873adf9673a
After: https://www.loom.com/share/0603d4f083b34f919aeaa0fe9f696ed1

Release note (ui change): The database details page now supports a
    large number of tables for a single database. Sort will be disabled
    if more than 40 tables are present in a database.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/103752)
<!-- Reviewable:end -->

maryliag reviewed (commented):
One small nit and a question about the node/regions filters

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/103752#-:-NW7x_tZBBeDXARV2W9X:b-iy2v8e)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j82w)

___
*[`pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsPage.tsx` line 272 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/103752#-NW7x4liAbVLfNR_oOfd:-NW7x4liAbVLfNR_oOfe:b927clt) ([raw file](https://github.com/cockroachdb/cockroach/blob/d714ba6dc0b1c978d95ff823ab8f84d4ae5696c3/pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsPage.tsx#L272)):*
> ```XML
>   ): boolean {
>     // If it's the first refresh or if it's been more than 5 minutes then
>     // do a refresh
> ```

nit: period on all comments (the same for the ones below)
___
*[`pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsPage.tsx` line 316 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/103752#-NW7xB1a7lJtHLWg9BVE:-NW7xB1a7lJtHLWg9BVF:bxrdsm8) ([raw file](https://github.com/cockroachdb/cockroach/blob/d714ba6dc0b1c978d95ff823ab8f84d4ae5696c3/pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsPage.tsx#L316)):*
> ```XML
>     this.setState({ ...this.state, lastRefreshTime: moment.utc() });
>     let lastDetailsError: Error;
>     // load everything by default
> ```

nit: Load
___
*[`pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsPage.tsx` line 389 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/103752#-NW7xKbf7983M9xnxw0Y:-NW7xKbg8NH4VZRVjqLp:b-ey9r41) ([raw file](https://github.com/cockroachdb/cockroach/blob/054f8dcd2a90bdd86a928157f499a72e75dc1b86/pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsPage.tsx#L389)):*
> ```XML
>       .filter(table => (search ? filterBySearchQuery(table, search) : true))
>       .filter(table => {
>         if (regionsSelected.length == 0 && nodesSelected.length == 0)
> ```

we also have filter by node/region. I assume you started you cluster with single node, so this is why you don't see that filter. Start something with several nodes and then make sure this part is still working
___
*[`pkg/ui/workspaces/cluster-ui/src/databasesPage/databasesPage.tsx` line 467 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/103752#-NW7xYR77jUvxBMKa5WN:-NW7xYR85k0SuD9C5J2O:b80nk5s) ([raw file](https://github.com/cockroachdb/cockroach/blob/d714ba6dc0b1c978d95ff823ab8f84d4ae5696c3/pkg/ui/workspaces/cluster-ui/src/databasesPage/databasesPage.tsx#L467)):*
> ```XML
>     prevProps: Readonly<DatabasesPageProps>,
>   ): boolean {
>     // If it's the first refresh or if it's been more than 5 minutes then
> ```

nit: period on comments


<!-- Sent from Reviewable.io -->

THardy98 reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/103752)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j82w)

___
*[`pkg/ui/workspaces/cluster-ui/src/api/databaseDetailsApi.ts` line 504 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/103752#-NW7yNnL-qgWcFSa1PR8:-NW7yNnL-qgWcFSa1PR9:bntaeet) ([raw file](https://github.com/cockroachdb/cockroach/blob/d714ba6dc0b1c978d95ff823ab8f84d4ae5696c3/pkg/ui/workspaces/cluster-ui/src/api/databaseDetailsApi.ts#L504)):*
> ```XML
> }
> 
> function createSqlExecutionRequest(
> ```

Might be a good idea to move this to `sqlApi.ts`, we can use this elsewhere
___
*[`pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsPage.tsx` line 276 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/103752#-NW7xlRC8qSBEHa3mvhc:-NW7xlRC8qSBEHa3mvhd:b7aocsf) ([raw file](https://github.com/cockroachdb/cockroach/blob/d714ba6dc0b1c978d95ff823ab8f84d4ae5696c3/pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsPage.tsx#L276)):*
> ```XML
>       !this.state.lastRefreshTime ||
>       this.state.lastRefreshTime <
>         moment.utc().subtract(moment.duration(5, "m"))
> ```

What's the need to poll database details every 5 minutes?

Currently it's still a pretty expensive API call and I doubt much changes within 5 minute intervals to warrant refreshes.


<!-- Sent from Reviewable.io -->

maryliag reviewed (commented):
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/103752)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j82w)



<!-- Sent from Reviewable.io -->

j82w commented:
*[`pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsPage.tsx` line 276 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/103752#-NW7xlRC8qSBEHa3mvhc:-NW8DaI83wHHGPIQQ17-:bt9nt9s) ([raw file](https://github.com/cockroachdb/cockroach/blob/d714ba6dc0b1c978d95ff823ab8f84d4ae5696c3/pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsPage.tsx#L276)):*
<details><summary><i>Previously, THardy98 (Thomas Hardy) wrote…</i></summary><blockquote>

What's the need to poll database details every 5 minutes?

Currently it's still a pretty expensive API call and I doubt much changes within 5 minute intervals to warrant refreshes.
</blockquote></details>

The db pages are still automatic refresh. I went ahead and removed it.


<!-- Sent from Reviewable.io -->

Commit:
ui: database details page now supports a large number of tables
Previos design:
1. The ui would attempt to a network call for all tables in a db.
2. The ui would completly fail if more than roughly 900 tables were
    present in a db. The number could be less if the table names were
    long.

New design:
1. The ui will only load the table details of the first 2 pages. This
    is a currently set to 20 tables per a page. If more than 40 tables
    are present in a db, the ui will disable sort. This is necessary
    because it would require loading the details of all tables which
    results in to many network calls overloading the db. The user can
    still filter using the table name.
2. The ui will now load all the table names. If a max size error is hit
    the ui will loop until it pulls all the table names.

Fixes: #98819
Epic: none

Release note (ui change): The database details page now supports a
    large number of tables for a single database. Sort will be disabled
    if more than 40 tables are present in a database.

j82w commented:
*[`pkg/ui/workspaces/cluster-ui/src/api/databaseDetailsApi.ts` line 504 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/103752#-NW7yNnL-qgWcFSa1PR8:-NW90bOBBC2Bgf5RzR_y:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/d714ba6dc0b1c978d95ff823ab8f84d4ae5696c3/pkg/ui/workspaces/cluster-ui/src/api/databaseDetailsApi.ts#L504)):*
<details><summary><i>Previously, THardy98 (Thomas Hardy) wrote…</i></summary><blockquote>

Might be a good idea to move this to `sqlApi.ts`, we can use this elsewhere
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

THardy98 reviewed (approved):
Nice improvement 
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/103752#-:-NW9swO4-V2_oDIldoMG:bas2eso)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @maryliag)



<!-- Sent from Reviewable.io -->

maryliag reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 4 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/103752#-:-NWCzGCS7l0ZmN1AYwLw:bnfp4nl)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @j82w)



<!-- Sent from Reviewable.io -->

j82w commented:
bors r+
